From 1a942e318dcdc20b77d0ea164db5efebead7b943 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 28 May 2014 16:28:14 +0800
Subject: [PATCH 2/2] ENGR00307764 wifi:fix first connection failure issue in
 wfd

Asynchronization of sink's getting IP and dnsmasq handshake would result
in first connection failure randomly in wifi display case. And property to
sync it will guarantee the sequence right.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 src/dnsmasq.c | 1 +
 src/dnsmasq.h | 2 +-
 src/rfc2131.c | 5 ++---
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/dnsmasq.c b/src/dnsmasq.c
index 2f0fbb6..a5badb2 100755
--- a/src/dnsmasq.c
+++ b/src/dnsmasq.c
@@ -96,6 +96,7 @@ int main (int argc, char **argv)
   textdomain("dnsmasq");
 #endif
 
+  property_set("net.dnsmasq.completed", "0");
   sigact.sa_handler = sig_handler;
   sigact.sa_flags = 0;
   sigemptyset(&sigact.sa_mask);
diff --git a/src/dnsmasq.h b/src/dnsmasq.h
index 0b9dbd2..726b32d 100755
--- a/src/dnsmasq.h
+++ b/src/dnsmasq.h
@@ -100,7 +100,7 @@
 #ifndef HAVE_LINUX_NETWORK
 #  include <net/if_dl.h>
 #endif
-
+#include "cutils/properties.h"
 #if defined(HAVE_LINUX_NETWORK)
 #include <linux/capability.h>
 /* There doesn't seem to be a universally-available 
diff --git a/src/rfc2131.c b/src/rfc2131.c
index 1ef8569..2abbf97 100755
--- a/src/rfc2131.c
+++ b/src/rfc2131.c
@@ -1217,9 +1217,8 @@ size_t dhcp_reply(struct dhcp_context *context, char *iface_name, int int_index,
 	    lease->override = override;
 	  else
 	    override = lease->override;
-
-	  log_packet("DHCPACK", &mess->yiaddr, emac, emac_len, iface_name, hostname, mess->xid);  
-	  
+	  property_set("net.dnsmasq.completed", "1");
+	  log_packet("DHCPACK", &mess->yiaddr, emac, emac_len, iface_name, hostname, mess->xid);
 	  clear_packet(mess, end);
 	  option_put(mess, end, OPTION_MESSAGE_TYPE, 1, DHCPACK);
 	  option_put(mess, end, OPTION_SERVER_IDENTIFIER, INADDRSZ, ntohl(server_id(context, override, fallback).s_addr));
-- 
1.8.0

